pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1642,18 +1642,33 @@ func createMirrorFilter(ctx configContext, filter *k8s.HTTPRequestMirrorFilter,
 	return &istio.HTTPMirrorPolicy{Destination: dst, Percentage: percent}, nil
 }
 
-func createRewriteFilter(filter *k8s.HTTPURLRewriteFilter) *istio.HTTPRewrite {
+func createRewriteFilter(matches []k8s.HTTPRouteMatch, filter *k8s.HTTPURLRewriteFilter) *istio.HTTPRewrite {
 	if filter == nil {
 		return nil
 	}
 	rewrite := &istio.HTTPRewrite{}
 	if filter.Path != nil {
 		switch filter.Path.Type {
 		case k8s.PrefixMatchHTTPPathModifier:
+			prefixMatchAll := false
+			for _, match := range matches {
+				if p := match.Path; p != nil {
+					if v := p.Value; v != nil && *v == "/" {
+						prefixMatchAll = true
+						break
+					}
+				}
+			}
 			rewrite.Uri = strings.TrimSuffix(*filter.Path.ReplacePrefixMatch, "/")
 			if rewrite.Uri == "" {
 				// `/` means removing the prefix
 				rewrite.Uri = "/"
+			} else if prefixMatchAll {
+				rewrite.UriRegexRewrite = &istio.RegexRewrite{
```

Comment:
So we're using regex matching over standard matching to get around envoy?

Reply:
yep. I don't love it -- we can see about getting afix into envoy itself

Reply:
Did you ever end up opening an issue in Envoy?

Reply:
No, pending decision in https://github.com/kubernetes-sigs/gateway-api/issues/3592